sample-api/api/src/Controllers/ToDoItemsController.cs

```diff
@@ -0,0 +1,147 @@
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Identity.Web.Resource;
+using ToDoApi.Models;
+
+namespace ToDoApi.Controllers
+{
+    //[Route("api/[controller]")]
+    [Authorize]
+    [RequiredScope("ToDoItem.ReadWrite")]
+    public class ToDoItemsController : ODataController
+    {
+        private readonly ToDoContext _context;
+        private readonly ILogger<ToDoItemsController> _logger;
+
+        public ToDoItemsController(ToDoContext context, ILogger<ToDoItemsController> logger)
+        {
+            _context = context;
+            _logger = logger;
+        }
+
+        // GET: /ToDoItems
+        [HttpGet]
+        [EnableQuery()]
+        public async Task<ActionResult<IQueryable<ToDoItem>>> GetToDoItemsAsync()
+        {
+            await LogRequestAsync(Request);
+            return _context.ToDoItems;
+        }
+
+        // GET: /ToDoItems/5
+        //[HttpGet("{key}")]
+        [HttpGet]
+        [EnableQuery()]
+        public async Task<ActionResult<ToDoItem>> GetToDoItemAsync(Guid key)
+        {
+            await LogRequestAsync(Request);
+            var todoItem = await _context.ToDoItems.FindAsync(key);
+
+            if (todoItem == null)
+            {
+                return NotFound();
+            }
+
+            return todoItem;
+        }
+
+        // PUT: /ToDoItems/5
+        //[HttpPatch("{id}")]
+        [HttpPatch]
+        public async Task<IActionResult> PatchToDoItemAsync(Guid key, Delta<ToDoItem> delta)
+        {
+            await LogRequestAsync(Request);
+            if (delta.GetChangedPropertyNames().Contains("Id"))
+            {
+                return BadRequest();
+            }
+
+            var item = await _context.ToDoItems.FindAsync(key);
+
+            if (item == null)
+            {
+                return NotFound();
+            }
+
+            delta.Patch(item);
+
+            _context.Update(item);
+
+            try
+            {
+                await _context.SaveChangesAsync();
+            }
+            catch (DbUpdateConcurrencyException)
+            {
+                if (!ToDoItemExists(key))
+                {
+                    return NotFound();
+                }
+                else
+                {
+                    throw;
+                }
+            }
+
+            return Updated(item);
+        }
+
+        // POST: /ToDoItems
+        [HttpPost]
+        public async Task<ActionResult<ToDoItem>> PostToDoItemAsync([FromBody]ToDoItem todoItem)
+        {
+            await LogRequestAsync(Request);
+            _context.ToDoItems.Add(todoItem);
+            await _context.SaveChangesAsync();
+
+            return Created(todoItem);
+            //return CreatedAtAction(nameof(GetToDoItem), new { id = todoItem.Id }, todoItem);
+        }
+
+        // DELETE: /ToDoItems/5
+        //[HttpDelete("{id}")]
+        [HttpDelete]
+        public async Task<IActionResult> DeleteToDoItemAsync(Guid key)
```

Comment:
from URL? or from patH?

Reply:
It's in the path: `DELETE /ToDoItems/{guid}`

Reply:
ok I though an attribute on the parameter was required, but apparently it maps the parameters automatically.
Can you remove the commented out attribute please?